File: src/geneticAlgorithm/species.cpp
#include <geneticAlgorithm/species.hpp>
#include "geneticAlgorithm/lifeform.hpp"
#include <geneticAlgorithm/geneticAlgorithm.hpp>
#include <simulator/simulator.hpp>

void Species::addCreature(LifeForm* creature) {
    members.push_back(creature);
};


void Species::removeCreature(LifeForm* creature) {
    members.erase(std::remove(members.begin(), members.end(), creature), members.end());
    if (members.empty()) {
        deathTime = Simulator::get().getStep();
    }
};

File: rust/src/genetic_algorithm/species.rs
pub struct Species {
  mascot_id: u32,
  members: Vec<u32>,
  origin_time: u32,
  death_time: u32,
}

impl Species {
  pub fn new(mascot_id: u32, origin_time: u32) -> Self {
    Self {
      mascot_id,
      members: Vec::new(),
      origin_time,
      death_time: 0,
    }
  }
}